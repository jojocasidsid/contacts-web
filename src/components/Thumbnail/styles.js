import styled from '@emotion/styled';
import PersonIcon from '@mui/icons-material/Person';

export const ThumbnailWrapper = styled('div')`
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 20px;
  flex-direction: column;
  border: 1px solid ${({ theme }) => theme.palette.primary.main};
`;

export const ActionWrapper = styled('div')`
  & > button {
    margin: 20px;
  }
`;

export const StyledImage = styled(PersonIcon)`
  font-size: 200px;
`;
