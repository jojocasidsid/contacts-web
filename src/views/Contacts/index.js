import React from 'react';
import ThumbnailList from 'components/ThumbnailList';
import useListingTable from 'hooks/useListingTable';
import { useQuery } from 'react-query';
import useDebounce from 'hooks/useDebounce';
import ContactService from 'services/Contacts';

const Contacts = () => {
  const { filters, setFilters } = useListingTable({
    search: '',
    itemsPerPage: 10,
    page: 1
  });

  const queryParams = useDebounce(filters, 500);

  const { isLoading, data } = useQuery(
    ['contacts', queryParams],
    () => {
      const mappedQueryParams = {
        _page: filters.page,
        _limit: filters.itemsPerPage,
        q: filters.search
      };

      console.log(mappedQueryParams);

      return ContactService.list(mappedQueryParams);
    },
    { refetchOnWindowFocus: true, refetchOnMount: true }
  );

  console.log(data);

  return (
    <div>
      <ThumbnailList
        setFilters={setFilters}
        search={filters.search}
        page={filters.page}
        totalPage={data?.totalPage}
        data={data?.data}
        isLoading={isLoading}
      />
    </div>
  );
};

export default Contacts;
