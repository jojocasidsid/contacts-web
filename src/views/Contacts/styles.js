import styled from '@emotion/styled';

export const ContactsWrapper = styled('div')``;
