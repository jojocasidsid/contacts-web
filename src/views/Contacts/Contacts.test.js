import React from 'react';
import renderer from 'react-test-renderer';
import { SnackbarProvider } from 'notistack';
import { ThemeProvider } from '@mui/material/styles';
import { QueryClient, QueryClientProvider } from 'react-query';

import theme from 'config/theme/theme';

import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';

import Contacts from '.';

const queryClient = new QueryClient();

describe('Contacts', () => {
  it('should render without crashing', () => {
    const tree = renderer.create(
      <ThemeProvider theme={theme}>
        <QueryClientProvider client={queryClient}>
          <SnackbarProvider
            maxSnack={5}
            anchorOrigin={{
              vertical: 'bottom',
              horizontal: 'right'
            }}
            preventDuplicate
            autoHideDuration={5000}
          >
            <Router>
              <Routes>
                <Route path="*" element={<Contacts />} />
              </Routes>
            </Router>
          </SnackbarProvider>
        </QueryClientProvider>
      </ThemeProvider>
    );

    expect(tree).toMatchSnapshot();
  });
});
