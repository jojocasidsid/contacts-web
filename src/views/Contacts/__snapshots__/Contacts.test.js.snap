// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contacts should render without crashing 1`] = `
@keyframes animation-0 {
  0% {
    -webkit-transform: rotate(0deg);
    -moz-transform: rotate(0deg);
    -ms-transform: rotate(0deg);
    transform: rotate(0deg);
  }

  100% {
    -webkit-transform: rotate(360deg);
    -moz-transform: rotate(360deg);
    -ms-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}

@keyframes animation-1 {
  0% {
    stroke-dasharray: 1px,200px;
    stroke-dashoffset: 0;
  }

  50% {
    stroke-dasharray: 100px,200px;
    stroke-dashoffset: -15px;
  }

  100% {
    stroke-dasharray: 100px,200px;
    stroke-dashoffset: -125px;
  }
}

.emotion-0 {
  background-color: #fff;
  color: #282B2F;
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 4px;
  box-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);
  padding: 20px;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  min-width: 0;
  padding: 0;
  margin: 0;
  border: 0;
  vertical-align: top;
  margin-bottom: 20px;
}

.emotion-3 .MuiInput-root {
  border: 1px solid #002F5D;
  border-radius: 2px;
  padding-left: 16px;
  min-width: 230px;
}

.emotion-3 .MuiInput-root input::-webkit-input-placeholder {
  font-weight: 600;
  font-size: 12px;
  font-style: normal;
}

.emotion-3 .MuiInput-root input::-moz-placeholder {
  font-weight: 600;
  font-size: 12px;
  font-style: normal;
}

.emotion-3 .MuiInput-root input:-ms-input-placeholder {
  font-weight: 600;
  font-size: 12px;
  font-style: normal;
}

.emotion-3 .MuiInput-root input::placeholder {
  font-weight: 600;
  font-size: 12px;
  font-style: normal;
}

.emotion-3 .MuiInputBase-root {
  margin-top: 23px;
}

.emotion-3 .MuiInputLabel-root {
  color: #303643;
  opacity: 0.5;
  font-size: 18px;
  font-weight: bold;
}

.emotion-3 .MuiInputLabel-root .Mui-focused {
  color: #303643;
}

.emotion-4 {
  color: #B5B5B5;
  font-family: Rubik;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.4375em;
  padding: 0;
  position: relative;
  display: block;
  transform-origin: top left;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  max-width: 133%;
  position: absolute;
  left: 0;
  top: 0;
  -webkit-transform: translate(0, -1.5px) scale(0.75);
  -moz-transform: translate(0, -1.5px) scale(0.75);
  -ms-transform: translate(0, -1.5px) scale(0.75);
  transform: translate(0, -1.5px) scale(0.75);
  -webkit-transition: color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,-webkit-transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,max-width 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,max-width 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
}

.emotion-4.Mui-focused {
  color: #FFFFFF;
}

.emotion-4.Mui-disabled {
  color: rgba(0, 0, 0, 0.38);
}

.emotion-4.Mui-error {
  color: #d32f2f;
}

.emotion-5 {
  font-family: Rubik;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.4375em;
  color: #282B2F;
  box-sizing: border-box;
  position: relative;
  cursor: text;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
}

.emotion-5.Mui-disabled {
  color: rgba(0, 0, 0, 0.38);
  cursor: default;
}

label+.emotion-5 {
  margin-top: 16px;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 0.01em;
  max-height: 2em;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  white-space: nowrap;
  color: rgba(0, 0, 0, 0.54);
  margin-right: 8px;
}

.emotion-7 {
  font: inherit;
  letter-spacing: inherit;
  color: currentColor;
  padding: 4px 0 5px;
  border: 0;
  box-sizing: content-box;
  background: none;
  height: 1.4375em;
  margin: 0;
  -webkit-tap-highlight-color: transparent;
  display: block;
  min-width: 0;
  width: 100%;
  -webkit-animation-name: mui-auto-fill-cancel;
  animation-name: mui-auto-fill-cancel;
  -webkit-animation-duration: 10ms;
  animation-duration: 10ms;
}

.emotion-7::-webkit-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-7::-moz-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-7:-ms-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-7::-ms-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-7:focus {
  outline: 0;
}

.emotion-7:invalid {
  box-shadow: none;
}

.emotion-7::-webkit-search-decoration {
  -webkit-appearance: none;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-7::-webkit-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-7::-moz-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-7:-ms-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-7::-ms-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-7:focus::-webkit-input-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-7:focus::-moz-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-7:focus:-ms-input-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-7:focus::-ms-input-placeholder {
  opacity: 0.42;
}

.emotion-7.Mui-disabled {
  opacity: 1;
  -webkit-text-fill-color: rgba(0, 0, 0, 0.38);
}

.emotion-7:-webkit-autofill {
  -webkit-animation-duration: 5000s;
  animation-duration: 5000s;
  -webkit-animation-name: mui-auto-fill;
  animation-name: mui-auto-fill;
}

.emotion-9 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  color: #fff;
  font-style: normal;
  font-weight: 600;
  font-size: 16x;
  line-height: 22px;
  letter-spacing: -0.02em;
  font-family: Rubik;
  text-transform: uppercase;
  min-width: 64px;
  padding: 6px 16px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  background-color: #002F5D;
  box-shadow: 0px 3px 1px -2px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12);
}

.emotion-9::-moz-focus-inner {
  border-style: none;
}

.emotion-9.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-9 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-9:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgb(0, 32, 65);
  box-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);
}

@media (hover: none) {
  .emotion-9:hover {
    background-color: #002F5D;
  }
}

.emotion-9:active {
  box-shadow: 0px 5px 5px -3px rgba(0,0,0,0.2),0px 8px 10px 1px rgba(0,0,0,0.14),0px 3px 14px 2px rgba(0,0,0,0.12);
}

.emotion-9.Mui-focusVisible {
  box-shadow: 0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12);
}

.emotion-9.Mui-disabled {
  color: #282B2F;
  box-shadow: none;
  background-color: rgba(0, 0, 0, 0.12);
}

.emotion-10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 60vh;
}

.emotion-11 {
  display: inline-block;
  color: #002F5D;
  -webkit-animation: animation-0 1.4s linear infinite;
  animation: animation-0 1.4s linear infinite;
}

.emotion-12 {
  display: block;
}

.emotion-13 {
  stroke: currentColor;
  stroke-dasharray: 80px,200px;
  stroke-dashoffset: 0;
  -webkit-animation: animation-1 1.4s ease-in-out infinite;
  animation: animation-1 1.4s ease-in-out infinite;
}

.emotion-14 {
  overflow: auto;
  color: #282B2F;
  font-size: 0.875rem;
}

.emotion-14:last-child {
  padding: 0;
}

.emotion-15 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 16px;
  padding-right: 16px;
  min-height: 56px;
  min-height: 52px;
  padding-right: 2px;
}

@media (min-width:600px) {
  .emotion-15 {
    padding-left: 24px;
    padding-right: 24px;
  }
}

@media (min-width:0px) and (orientation: landscape) {
  .emotion-15 {
    min-height: 48px;
  }
}

@media (min-width:600px) {
  .emotion-15 {
    min-height: 64px;
  }
}

@media (min-width:0px) and (orientation: landscape) {
  .emotion-15 {
    min-height: 52px;
  }
}

@media (min-width:600px) {
  .emotion-15 {
    min-height: 52px;
    padding-right: 2px;
  }
}

.emotion-15 .MuiTablePagination-actions {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  margin-left: 20px;
}

.emotion-16 {
  -webkit-flex: 1 1 100%;
  -ms-flex: 1 1 100%;
  flex: 1 1 100%;
}

.emotion-17 {
  color: #282B2F;
  font-style: normal;
  font-weight: 300;
  font-size: 14px;
  line-height: 24px;
  font-family: Rubik;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.emotion-18 {
  font-family: Rubik;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.4375em;
  color: #282B2F;
  box-sizing: border-box;
  position: relative;
  cursor: text;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: inherit;
  font-size: inherit;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  margin-right: 32px;
  margin-left: 8px;
}

.emotion-18.Mui-disabled {
  color: rgba(0, 0, 0, 0.38);
  cursor: default;
}

.emotion-18 .MuiTablePagination-select {
  padding-left: 8px;
  padding-right: 24px;
  text-align: right;
  -webkit-text-align-last: right;
  text-align-last: right;
}

.emotion-19 {
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border-radius: 0;
  cursor: pointer;
  font: inherit;
  letter-spacing: inherit;
  color: currentColor;
  padding: 4px 0 5px;
  border: 0;
  box-sizing: content-box;
  background: none;
  height: 1.4375em;
  margin: 0;
  -webkit-tap-highlight-color: transparent;
  display: block;
  min-width: 0;
  width: 100%;
  -webkit-animation-name: mui-auto-fill-cancel;
  animation-name: mui-auto-fill-cancel;
  -webkit-animation-duration: 10ms;
  animation-duration: 10ms;
}

.emotion-19:focus {
  background-color: rgba(0, 0, 0, 0.05);
  border-radius: 0;
}

.emotion-19::-ms-expand {
  display: none;
}

.emotion-19.Mui-disabled {
  cursor: default;
}

.emotion-19[multiple] {
  height: auto;
}

.emotion-19:not([multiple]) option,
.emotion-19:not([multiple]) optgroup {
  background-color: #fff;
}

.emotion-19.emotion-19.emotion-19 {
  padding-right: 24px;
  min-width: 16px;
}

.emotion-19.MuiSelect-select {
  height: auto;
  min-height: 1.4375em;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
}

.emotion-19::-webkit-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-19::-moz-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-19:-ms-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-19::-ms-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-19:focus {
  outline: 0;
}

.emotion-19:invalid {
  box-shadow: none;
}

.emotion-19::-webkit-search-decoration {
  -webkit-appearance: none;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-19::-webkit-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-19::-moz-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-19:-ms-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-19::-ms-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-19:focus::-webkit-input-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-19:focus::-moz-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-19:focus:-ms-input-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-19:focus::-ms-input-placeholder {
  opacity: 0.42;
}

.emotion-19.Mui-disabled {
  opacity: 1;
  -webkit-text-fill-color: rgba(0, 0, 0, 0.38);
}

.emotion-19:-webkit-autofill {
  -webkit-animation-duration: 5000s;
  animation-duration: 5000s;
  -webkit-animation-name: mui-auto-fill;
  animation-name: mui-auto-fill;
}

.emotion-20 {
  bottom: 0;
  left: 0;
  position: absolute;
  opacity: 0;
  pointer-events: none;
  width: 100%;
  box-sizing: border-box;
}

.emotion-21 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
  position: absolute;
  right: 0;
  top: calc(50% - .5em);
  pointer-events: none;
  color: rgba(0, 0, 0, 0.54);
}

.emotion-21.Mui-disabled {
  color: #282B2F;
}

.emotion-22 {
  color: #282B2F;
  font-style: normal;
  font-weight: 300;
  font-size: 14px;
  line-height: 24px;
  font-family: Rubik;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.emotion-23 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  text-align: center;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  font-size: 1.5rem;
  padding: 8px;
  border-radius: 50%;
  overflow: visible;
  color: rgba(0, 0, 0, 0.54);
  -webkit-transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  color: inherit;
}

.emotion-23::-moz-focus-inner {
  border-style: none;
}

.emotion-23.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-23 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-23:hover {
  background-color: rgba(0, 0, 0, 0.04);
}

@media (hover: none) {
  .emotion-23:hover {
    background-color: transparent;
  }
}

.emotion-23.Mui-disabled {
  background-color: transparent;
  color: #282B2F;
}

.emotion-24 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
}

<div>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 emotion-0"
  >
    <div
      className="MuiBox-root emotion-1"
    >
      <div
        className="MuiBox-root emotion-2"
      >
        <div
          className="MuiFormControl-root MuiTextField-root emotion-3"
        >
          <label
            className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorInfo emotion-4"
            data-shrink={true}
            htmlFor="searchBar"
            id="searchBar-label"
            style={
              Object {
                "color": "#303643",
              }
            }
          >
            Search
          </label>
          <div
            className="MuiInput-root MuiInputBase-root MuiInputBase-colorInfo MuiInputBase-formControl MuiInputBase-adornedStart emotion-5"
            onClick={[Function]}
          >
            <div
              className="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-standard MuiInputAdornment-sizeMedium emotion-6"
            >
              <span
                className="notranslate"
              >
                ​
              </span>
              <img
                alt="search"
                src="/assets/search.svg"
              />
            </div>
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInput-input MuiInputBase-input MuiInputBase-inputAdornedStart emotion-7"
              disabled={false}
              id="searchBar"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="search"
              required={false}
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        className="MuiBox-root emotion-2"
      >
        <button
          className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  emotion-9"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          Add Contact
        </button>
      </div>
    </div>
    <div
      className="MuiBox-root emotion-10"
    >
      <span
        className="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary emotion-11"
        role="progressbar"
        style={
          Object {
            "height": 40,
            "width": 40,
          }
        }
      >
        <svg
          className="MuiCircularProgress-svg emotion-12"
          viewBox="22 22 44 44"
        >
          <circle
            className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate emotion-13"
            cx={44}
            cy={44}
            fill="none"
            r={20.2}
            strokeWidth={3.6}
            style={Object {}}
          />
        </svg>
      </span>
    </div>
    <div
      className="MuiTablePagination-root emotion-14"
    >
      <div
        className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular MuiTablePagination-toolbar emotion-15"
      >
        <div
          className="MuiTablePagination-spacer emotion-16"
        />
        <p
          className="MuiTablePagination-selectLabel emotion-17"
        >
          Rows per page:
        </p>
        <div
          className="MuiInputBase-root MuiInputBase-colorPrimary emotion-18"
          onClick={[Function]}
          variant="standard"
        >
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            className="MuiTablePagination-select MuiSelect-select MuiSelect-standard MuiInputBase-input emotion-19"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            role="button"
            tabIndex={0}
          >
            10
          </div>
          <input
            aria-hidden={true}
            className="MuiSelect-nativeInput emotion-20"
            onAnimationStart={[Function]}
            onChange={[Function]}
            tabIndex={-1}
            value={10}
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTablePagination-selectIcon MuiSelect-icon MuiSelect-iconStandard emotion-21"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
        <p
          className="MuiTablePagination-displayedRows emotion-22"
        >
          0–0 of 0
        </p>
        <div
          className="MuiTablePagination-actions"
        >
          <button
            aria-label="Go to previous page"
            className="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium emotion-23"
            disabled={true}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={-1}
            title="Go to previous page"
            type="button"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-24"
              data-testid="KeyboardArrowLeftIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
              />
            </svg>
          </button>
          <button
            aria-label="Go to next page"
            className="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium emotion-23"
            disabled={true}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={-1}
            title="Go to next page"
            type="button"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-24"
              data-testid="KeyboardArrowRightIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
